meson.build

```
@@ -0,0 +1,52 @@
+project('arx', 'cpp', 'c',
+  license : 'Apache-2.0',
+  version : '1.4.0',  # semantic-release
+  default_options : [
+    'warning_level=3',
+    'cpp_std=c++17',
+  ]
+)
+
+cxx = meson.get_compiler('cpp')
+
+deps = [
+  dependency('arrow'),
+  dependency('llvm', version : '>=13.0.0'),
+  dependency('CLI11'),
+  dependency('glog'),
+]
+
+inc = include_directories('arx/include')
+
+sources = [
+  'arx/src/input.cpp',
+  'arx/src/lexer.cpp',
+  'arx/src/parser.cpp',
+  'arx/src/error.cpp',
+  'arx/src/codegen.cpp',
+  'arx/src/settings.cpp',
+  'arx/src/utils.cpp',
+]
+
+if get_option('tests').enabled()
+  deps += [dependency('gtest')]
```

I would instead use:

```suggestion
gtest_dep = dependency('gtest', required: get_option('tests'))
if gtest_dep.found()
    deps += [gtest_dep]
```

The point of a feature option is to be a tristate option. If:
- enabled, then that's equivalent to default "required: true".
- disabled, then the dependency logs as "skipped: feature **tests** is disabled"
- auto, then that's equivalent to "required: false"

So, the effect is identical to what you currently did, except that in the auto case it will try to find gtest, and only enable the tests if gtest is found.

---

That looks very nice. Thanks for the review :) I will apply your suggestion soon!! Thanks